doc/user_guide/Generators.ipynb

```diff
@@ -0,0 +1,340 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f62c5bb1-91c5-4663-b429-d45f4a55b344",
+   "metadata": {},
+   "source": [
+    "# Generators\n",
+    "\n",
+    "As described in the [Dynamic Parameters](./Dynamic_Parameters.ipynb) user guide Param offers a variety of programming models. While Dynamic parameters allow you to build applications in a \"pull\" based paradigm, dependencies and watchers allow setting up a \"push\" based model where data flows through the application when an event is triggered. How such events are triggered is largely dependent on the use case, e.g. often such events will eminate from a UI that updates parameters, however generators can be another source of event generation and offer an easy and flexible approach for generating data that arrives from an external source.\n",
+    "\n",
+    "Let's start with a simple but concrete example. Let's say we are observing an external sensor measuring the output of a tidal gauge:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7dd8369e-631b-443d-a266-dc96745f8f4b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "\n",
+    "last = 0\n",
+    "\n",
+    "def tidal_gauge():\n",
+    "    global last\n",
+    "    last += random.gauss(0, 1)/2.\n",
+    "    return last\n",
+    "\n",
+    "tidal_gauge()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e8361a2-8af5-4fd0-ba42-646c8e988f87",
+   "metadata": {},
+   "source": [
+    "Instead of scheduling some periodic event on some event loop we can simply write a generator function to poll his sensor at some defined interval:"
```

Comment:
```suggestion
    "Instead of scheduling some periodic event on some event loop we can simply write a generator function to poll this sensor at some defined interval:"
```

Reply:
Also too many "some"s in the same sentence!